src/protobuf-net.BuildTools/Internal/CodeGen/Parsers/ServiceMethodCodeGenModelParser.cs

```diff
@@ -46,11 +84,15 @@ private CodeGenServiceMethod.Type ParseResponseType(IMethodSymbol symbol)
         };
     }
 
+    private bool IsGrpcCallContext(IParameterSymbol parameterSymbol) => parameterSymbol.Type.ToString() == "ProtoBuf.Grpc.CallContext" && parameterSymbol.Type.InProtoBufNamespace();
+    
+    private bool IsCancellationToken(IParameterSymbol parameterSymbol) => parameterSymbol.Type.ToString() == "System.Threading.CancellationToken" && parameterSymbol.Type.InThreadingNamespace();
+
     private CodeGenTypeRepresentation DetermineGenericTypeRepresentation(INamedTypeSymbol symbol) => symbol.ToString() switch
     {
-        "System.Collections.Generic.IAsyncEnumerable<T>" => CodeGenTypeRepresentation.AsyncEnumerable,
-        "System.Threading.Tasks.Task<T>" or "System.Threading.Tasks.Task<TResult>" => CodeGenTypeRepresentation.Task,
-        "System.Threading.Tasks.ValueTask<T>" or "System.Threading.Tasks.ValueTask<TResult>" => CodeGenTypeRepresentation.ValueTask,
+        "System.Collections.Generic.IAsyncEnumerable<T>" when symbol.InGenericCollectionsNamespace() => CodeGenTypeRepresentation.AsyncEnumerable,
```

Comment:
if we're testing the namespace here, surely we can just look at `symbol.Name` on the left-hand-side?